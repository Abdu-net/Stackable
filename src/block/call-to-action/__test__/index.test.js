import { edit, save } from '../index'

describe( 'Call to Action', () => {
	test( 'block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				url: 'http://www.gambit.ph/',
				ctaTitle: 'Get Started Today',
				bodyText: 'Get Stackable: Ultimate Gutenberg Blocks today.  Apart from adding new blocks, it gives Gutenberg users more options and settings to tinker with, expanding Gutenberg’s functionality.',
				buttonText: 'Button',
				color: '#2091e1',
				textColor: '#ffffff',
				titleColor: '#23282d',
				bodyTextColor: '#ffffff',
				bgColor: '#f4f4f4',
				size: 'normal',
				borderButtonRadius: 4,
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				url: 'http://www.gambit.ph/',
				ctaTitle: 'Get Started Today',
				bodyText: 'Get Stackable: Ultimate Gutenberg Blocks today.  Apart from adding new blocks, it gives Gutenberg users more options and settings to tinker with, expanding Gutenberg’s functionality.',
				buttonText: 'Button',
				color: '#2091e1',
				textColor: '#ffffff',
				titleColor: '#23282d',
				bodyTextColor: '#ffffff',
				bgColor: '#f4f4f4',
				size: 'normal',
				borderButtonRadius: 4,
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
