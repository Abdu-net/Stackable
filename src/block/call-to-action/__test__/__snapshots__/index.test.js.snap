// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Call to Action block edit matches snapshots 1`] = `
<Fragment>
  <InspectorControls>
    <PanelBody>
      <SelectControl
        label="Button Size"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Small",
              "value": "small",
            },
            Object {
              "label": "Normal",
              "value": "normal",
            },
            Object {
              "label": "Medium",
              "value": "medium",
            },
            Object {
              "label": "Large",
              "value": "large",
            },
          ]
        }
        value="normal"
      />
      <RangeControl
        label="Button Border Radius"
        max="50"
        min="1"
        onChange={[Function]}
        value={4}
      />
    </PanelBody>
    <PanelColorSettings
      colorSettings={
        Array [
          Object {
            "label": "Background Color",
            "onChange": [Function],
            "value": "#f4f4f4",
          },
          Object {
            "label": "Title Color",
            "onChange": [Function],
            "value": "#23282d",
          },
          Object {
            "label": "Body Text Color",
            "onChange": [Function],
            "value": "#ffffff",
          },
        ]
      }
      initialOpen={false}
      title="Color Settings"
    />
    <PanelColorSettings
      colorSettings={
        Array [
          Object {
            "label": "Button Background Color",
            "onChange": [Function],
            "value": "#2091e1",
          },
          Object {
            "label": "Button Text Color",
            "onChange": [Function],
            "value": "#ffffff",
          },
        ]
      }
      title="Button Colors"
    />
  </InspectorControls>
  <div
    className="ugb-cta"
    style={
      Object {
        "backgroundColor": "#f4f4f4",
      }
    }
  >
    <RichText
      className="ugb-cta-title"
      keepPlaceholderOnFocus={true}
      onChange={[Function]}
      placeholder="Add Title"
      style={
        Object {
          "color": "#23282d",
        }
      }
      tagName="h3"
      value="Get Started Today"
    />
    <RichText
      className="ugb-cta-bodyText"
      onChange={[Function]}
      placeholder="Write body text…"
      style={
        Object {
          "color": "#ffffff",
        }
      }
      tagName="p"
      value="Get Stackable: Ultimate Gutenberg Blocks today.  Apart from adding new blocks, it gives Gutenberg users more options and settings to tinker with, expanding Gutenberg’s functionality."
    />
    <ButtonEdit
      backgroundColor="#2091e1"
      borderRadius={4}
      color="#ffffff"
      onChange={[Function]}
      size="normal"
      text="Button"
    />
  </div>
</Fragment>
`;

exports[`Call to Action block save matches snapshots 1`] = `
<div
  className="ugb-cta"
  style={
    Object {
      "backgroundColor": "#f4f4f4",
    }
  }
>
  <Unknown
    className="ugb-cta-title"
    style={
      Object {
        "color": "#23282d",
      }
    }
    tagName="h3"
    value="Get Started Today"
  />
  <Unknown
    className="ugb-cta-bodyText"
    style={
      Object {
        "color": "#ffffff",
      }
    }
    tagName="p"
    value="Get Stackable: Ultimate Gutenberg Blocks today.  Apart from adding new blocks, it gives Gutenberg users more options and settings to tinker with, expanding Gutenberg’s functionality."
  />
  <ButtonSave
    backgroundColor="#2091e1"
    borderRadius={4}
    color="#ffffff"
    size="normal"
    text="Button"
    url="http://www.gambit.ph/"
  />
</div>
`;
